llvm/labs/lab2/mortina_nastya/Instrument_func_wrapper.cpp

```diff
@@ -0,0 +1,72 @@
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/raw_ostream.h"
+
+namespace {
+struct FunctionInstrumentation : llvm::PassInfoMixin<FunctionInstrumentation> {
+  llvm::PreservedAnalyses run(llvm::Function &func,
+                              llvm::FunctionAnalysisManager &) {
+    llvm::LLVMContext &ctx = func.getContext();
+    llvm::IRBuilder<> builder(ctx);
+    llvm::Module *mod = func.getParent();
+    bool hasStart = false;
+    bool hasEnd = false;
+
+    llvm::FunctionType *funcType =
+        llvm::FunctionType::get(llvm::Type::getVoidTy(ctx), false);
+    llvm::FunctionCallee startInstr =
+        (*mod).getOrInsertFunction("start_instrument", funcType);
+    llvm::FunctionCallee endInstr =
+        (*mod).getOrInsertFunction("end_instrument", funcType);
+
+    for (auto &blk : func) {
+      for (auto &instr : blk) {
+        if (llvm::isa<llvm::CallInst>(&instr)) {
+          llvm::CallInst *callInstruction = llvm::cast<llvm::CallInst>(&instr);
+          if (callInstruction->getCalledFunction() == startInstr.getCallee()) {
```

Comment:
Could you please check a function call presence via iterating through users of `start_instrument`/`end_instrument` functions?